File: python/238_Product of Array Except Self.py
#!/usr/bin/env python2

class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        hasZero = 0
        mul = 1
        for num in nums:
            if num != 0:
                mul *= num
            else:
                hasZero += 1

        if hasZero > 1:
            for i in range(len(nums)):
                nums[i] = 0
        elif hasZero == 1:
            for i in range(len(nums)):
                if nums[i] == 0:
                    nums[i] = mul
                else:
                    nums[i] = 0
        else:
            for i in range(len(nums)):
                nums[i] = mul/nums[i]
        return nums

File: java/src/brand_new/_238_Product_of_Array_Except_Self.java
package brand_new;

public class _238_Product_of_Array_Except_Self {
	public int[] productExceptSelf(int[] nums) {
		int mul = 1;
		int zero = 0;
		for (int i : nums) {
			if (i == 0) {
				zero++;
			} else {
				mul *= i;
			}
		}

		if (zero > 1) {
			for (int i = 0; i < nums.length; i++) {
				nums[i] = 0;
			}
		} else if (zero == 1) {
			for (int i = 0; i < nums.length; i++) {
				if (nums[i] == 0)
					nums[i] = mul;
				else
					nums[i] = 0;

			}
		} else {
			for (int i = 0; i < nums.length; i++) {
				nums[i] = mul / nums[i];
			}
		}
		return nums;
	}
}
